### PR TITLE
Pull Request for Initial QoD Cucumber Test contribution.

### PR DESCRIPTION
**Pull request by mdomale**
Pull Request for Initial QoD Cucumber Test contribution.
Fixes https://github.com/camaraproject/QualityOnDemand/issues/133

### CODE REVIEW

**Review by akoshunyadi — APPROVED** [review]

**Review by jlurien — COMMENTED** [review]
I think the first step would be to define the scope of the test spec in Gherkin (the *.,feature files), what aspects of the API must be covered and the level of detail of the tests. There are several frameworks and languages to implement the test steps, not sure if we need to consider just one. To me this is similar to the discussion we had about the convenience to have PI code contributions along with the spec or in separate repo. 

Also, if the objective of the test plan is to validate some implementation, we have to consider which parameters are environment dependent (Urls, MSISDNs, IP addresses, etc) and how to reflect that in the test plan.

File: code/cucumber/src/test/java/stepDefinitions/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package stepDefinitions;
+
+public class AppConfig {
```
[the remark this answers is not in the document]
Comment (mdomale) [reply code/cucumber/src/test/java/stepDefinitions/AppConfig.java:27]: As agreed in the issue https://github.com/camaraproject/WorkingGroups/issues/61  this is the proposal using the toolstack Maven,Junit 5,Java,wiremock  .It is not depending on the implementation, that's why it is committed here and not in the PI.

[the remark this answers is not in the document]
Comment (mdomale) [reply code/cucumber/src/test/java/stepDefinitions/AppConfig.java:27]: The Parameter values present in AppConfig.java are not environment specific as they are defined considering qod service is up and running locally and are used as test data only. Additionally the service and SCEF URLs should be changed on demand, but the test data has no external dependency.

**Comment by mdomale** [comment]
@patrice-conil  Intention for this proposal is having basic tests for QoD using cucumber framework. These test cases can be executed locally without any need of dependencies except QoD service up and running  on local. As these are basic tests and easily adaptable by everyone for getting basic confidence around QoD service workability , on side note these can be extended further as well.
Full qualification tests are not intended as a part of this contribution . By using cucumber framework we have flexibility to switch to any required backend . cc @akoshunyadi

**Review by patrice-conil — APPROVED** [review]
Fine for me

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Comment by hdamker** [comment]
Thanks for the reviews @akoshunyadi @patrice-conil @shilpa-padgaonkar

**Pull request merged**